### DISCUSSION TITLE
Suggestion on a way to stream output from a watch command

### DISCUSSION THREAD

**Discussion by rkpatel33** [Textualize/textual#245] [Q&A]
I am trying to setup a widget that will show continuous output from a Kubernetes watch command or log follow command:

```sh
kubectl get events --watch
kubectl logs -f -l app=main --all-containers
```
I am trying to use Python subprocess to capture these logs and then `render` them on a 2 sec refresh, but I am having trouble with figuring out how to implement this in a `Widget`. All the methods I've tried result in a loop where the process never ends waiting for the next line, so you never get to the end of the `render()` method to return a chunk of output.

**Comment by mwalkerr** [comment]
@rkpatel33 I haven't programmed in Python in a while, so take this with a grain of salt, but what I think you'd need to do is this: create a thread that reads from that process and writes to a queue (or buffer?). Then create a timer which reads from the queue if it's not empty and renders all the new content in the widget

**Comment by rkpatel33** [comment]
Thank you for the suggestion, I will look into that approach.

**Reply by mwlayne** [reply]
Did you ever solve this issue? I am thinking about doing something similar

**Comment by sminot** [comment]
I just saw this and thought it might help. 
https://github.com/Textualize/textual#timers-and-intervals
I can imagine your using a timer to periodically update the display from the stream.

**Comment by rkpatel33** [comment]
I did not, have not had a chance to return. I am using straight intervals
for certain panes that are refreshing a finite set amount of data.
